content/blog/2020/05/chatops_vmware_host_log_collection.md

```diff
@@ -0,0 +1,322 @@
+  "implementations": [
+    {"name": "vmware_generate_and_upload_log_bundle.sh", "requirements": ["shell"]}
+  ],
+  "parameters": {
+    "esx_username": {
+      "description": "Username to login to the esx host. Generally root.",
+      "type": "String[1]"
+    },
+    "esx_password": {
+      "description": "Password to login to the esx host. Generally root password.",
+      "type": "String[1]",
+      "sensitive": true
+    },
+    "esx_host": {
+      "description": "Esx host to collect log bundle from.",
+      "type": "String[1]"
+    },
+    "download_directory": {
+      "description": "Location in which the .zip is downloaded to from the esx host. Default /tmp/",
+      "type": "Optional[String]"
+    },
+    "sftp_site": {
+      "description": "URL to the SFTP site.",
+      "type": "String[1]"
+    },
+    "sftp_username": {
+      "description": "Username to login to the SFTP site.",
+      "type": "String[1]"
+    },
+    "sftp_password": {
+      "description": "Password to login to the SFTP site.",
+      "type": "String[1]",
+      "sensitive": true
+    },
+    "vmware_sr_number": {
+      "description": "Must be 11 digit VMware SR #. Directory on the SFTP server the log will upload to.",
+      "type": "String[1]"
+    }
+  }
+}
+```
+
+### Stackstorm Code
+
+Now that we have a functional Bolt task to do want we want, we can simply roll it into a Stackstorm workflow that can be called via slack.
+
+
+This Stackstorm action calls the bolt task above: vmware_host_collect_and_upload_log_bundle.yaml
+
+```
+---
+description: "Gather log bundle from specified ESX host and upload that log bundle to a VMware SR #"
+enabled: true
+runner_type: orquesta
+entry_point: workflows/vmware_host_collect_and_upload_log_bundle.yaml
+name: vmware_host_collect_and_upload_log_bundle
+pack: encore
+parameters:
+  vmware_sr_number:
+    type: string
+    description: "11 digit VMware SR number"
+    required: true
+  esx_host:
+    type: string
+    description: "FQDN or IP of ESX host to collect logs from"
+    required: true
+  download_server:
+    type: string
+    description: "FQDN or IP of Linux system to temporarily place logs for upload to support"
+    default: "{{ st2kv.system.stackstorm.server }}"
+```
+
+
+
+
+This Stackstorm workflow calls the Stackstorm action above: vmware_host_collect_and_upload_log_bundle.yaml
+
+One of the key things here is that environment variables are being populated for the usernames and passwords being used. This means that they are NOT visible on the system running this code.
+```
+---
+version: 1.0
+
+description: Workflow to collect and upload a log bundle from a specified ESX host
+
+input:
+  - vmware_sr_number
+  - esx_host
+  - download_server
+
+vars:
+  - action_error: ''
+  - action_result: ''
+
+output:
+  - action_error: "{{ ctx().action_error }}"
+  - action_result: "{{ ctx().action_result }}"
+
+tasks:
+  plan_run:
+    action: encore.bolt_plan
```

Comment:
others won't have access to this action/workflow

Reply:
The amount of code it would add to this post to add the action & workflow is pretty big, especially if we include the actions that bolt_plan uses.

I can add it if we want. I get that the code in the blog as-is won't work without it but i'm torn because it is so much.